src/send.c

```diff
@@ -117,7 +121,14 @@ static void send_type(const fastd_socket_t *sock, const fastd_peer_address_t *lo
 	msg.msg_control = cbuf;
 	msg.msg_controllen = 0;
 
+#ifdef __ANDROID__
+	/* PKTINFO will mess with Android VpnService.protect(socket) */
+	if (!conf.android_integration) {
+		add_pktinfo(&msg, local_addr);
+	}
+#else
```

Comment:
I think this check should be moved to `add_pktinfo` to avoid the `#else` branch.